File: app/src/main/java/ru/neosvet/ui/Tip.java
package ru.neosvet.ui;

import android.content.Context;
import android.os.Handler;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import java.util.Timer;
import java.util.TimerTask;

import ru.neosvet.vestnewage.R;

public class Tip {
    private View object;
    private boolean show = false;
    private Animation anShow, anHide;
    private Timer timer;

    final Handler hHide = new Handler(message -> {
        if (show)
            object.startAnimation(anHide);
        return false;
    });

    public Tip(Context context, final View object) {
        this.object = object;
        anShow = AnimationUtils.loadAnimation(context, R.anim.show);
        anHide = AnimationUtils.loadAnimation(context, R.anim.hide);
        anHide.setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {

            }

            @Override
            public void onAnimationEnd(Animation animation) {
                object.setVisibility(View.GONE);
                show = false;
            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });
    }

    public void show() {
        if (show)
            return;
        show = true;
        object.setVisibility(View.VISIBLE);
        object.startAnimation(anShow);
        timer = new Timer();
        timer.schedule(new TimerTask() {
            @Override
            public void run() {
                hHide.sendEmptyMessage(0);
            }
        }, 2500);
    }

    public void hide() {
        if (!show)
            return;
        timer.cancel();
        object.clearAnimation();
        object.setVisibility(View.GONE);
        show = false;
    }

    public void hideAnimated() {
        if (!show)
            return;
        timer.cancel();
        object.clearAnimation();
        object.startAnimation(anHide);
    }

    public boolean isShow() {
        return show;
    }
}

File: app/src/main/java/ru/neosvet/vestnewage/view/basic/Tip.kt
package ru.neosvet.vestnewage.view.basic

import android.content.Context
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import kotlinx.coroutines.*
import ru.neosvet.vestnewage.R

class Tip(context: Context?, private val view: View) {
    var isShow = false
        private set
    var autoHide = true
    private val anShow: Animation
    private val anHide: Animation
    private var job: Job? = null
    private val scope = CoroutineScope(Dispatchers.Default)

    init {
        anShow = AnimationUtils.loadAnimation(context, R.anim.show)
        anHide = AnimationUtils.loadAnimation(context, R.anim.hide)
        anHide.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationStart(animation: Animation) {}
            override fun onAnimationEnd(animation: Animation) {
                view.visibility = View.GONE
                isShow = false
            }

            override fun onAnimationRepeat(animation: Animation) {}
        })
    }

    fun show() {
        if (isShow) return
        isShow = true
        view.visibility = View.VISIBLE
        view.startAnimation(anShow)
        if (autoHide) {
            job?.cancel()
            job = scope.launch {
                delay(2500)
                if (isShow) view.post {
                    view.startAnimation(anHide)
                }
            }
        }
    }

    fun hide() {
        if (!isShow) return
        job?.cancel()
        view.clearAnimation()
        view.visibility = View.GONE
        isShow = false
    }
}